### PR TITLE
WIP: SLM code to extract footprint / central pixel

### PR DESCRIPTION
**Pull request by poilvert**
cc: @npinto

### CODE REVIEW

File: sthor/model/slm.py [hunk 1]
```diff
@@ -34,6 +36,95 @@ def __init__(self, in_shape, description):
...
+        h, w = self.in_shape[:2]
+
+        for parameters in layer_param[:nmax]:
+            nbh, nbw, s = parameters
+            h = 1 + (h - nbh) / s
+            w = 1 + (w - nbw) / s
+
+        return (h, w)
+
+    def rcp_field_central_px_coords(self,
```
Comment (npinto) [inline sthor/model/slm.py:82]: Let's ditch that ;-), and "replay" the operations with `view_as_windows`, `resample`, and "striding", propagating the `x_min`, `x_max`, `y_min`, `y_max`:
then we can use the slm model like this:

```
# take layer "1" with lpool
x_min = slm.layers[1][1]['x_min']
x_max = slm.layers[1][1]['x_max']

# center of the RF
xc = x_max - x_min / 2
```

**Comment by npinto** [comment]
API proposal to allow "bottom-up" / "top-down" RF ensembles

```
output_x_min, output_x_max, output_y_min, output_y_max = slm.layers[1][1]['input']
output_x_min.shape == input.shape
...
```

**Comment by npinto** [comment]
How should we proceed with this one? Should we postpone?

**Comment by poilvert** [comment]
yes I suspect we should. I am not sure that we have come across
most use cases thus far. When we will have a better overview of
things, we will then implement and merge. what do you think ?

On Wed, Apr 11, 2012 at 3:18 PM, Nicolas Pinto <
reply@reply.github.com

> wrote:
> 
> How should we proceed with this one? Should we postpone?
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/nsf-ri-ubicv/sthor/pull/5#issuecomment-5076688

## 

Nicolas POILVERT, PhD
Post-doctoral Fellow
Rowland Institute at Harvard

**Comment by npinto** [comment]
should we close this one?

**Comment by poilvert** [comment]
I guess we should, since the latest additions to the SLM
render this code kind of useless.

On Mon, Apr 16, 2012 at 12:07 PM, Nicolas Pinto <
reply@reply.github.com

> wrote:
> 
> should we close this one?
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/nsf-ri-ubicv/sthor/pull/5#issuecomment-5155822

## 

Nicolas POILVERT, PhD
Post-doctoral Fellow
Rowland Institute at Harvard

**Pull request closed without merging**